setup.py

```diff
@@ -1,65 +1,61 @@
+import io
 import platform
 import os
 from setuptools import setup
-from buildingspy import __version__
+
 # Python setup file.
 # See http://packages.python.org/an_example_pypi_project/setuptools.html
 
+MAIN_PACKAGE = 'buildingspy'
+PACKAGE_PATH =  os.path.abspath(os.path.join(os.path.dirname(__file__), MAIN_PACKAGE))
 
-def read(fname):
-    return open(os.path.join(os.path.dirname(__file__), fname)).read()
-
+# Version.
+version_path = os.path.join(PACKAGE_PATH, 'VERSION')
+with open(version_path) as f:
+    VERSION = f.read().strip()
 
-os_name = platform.system()
-lib_data = 'funnel/lib'
-if os_name == 'Windows':
-    lib_data = '{}/win64/*.dll'.format(lib_data)
-elif os_name == 'Linux':
-    lib_data = '{}/linux64/*.so'.format(lib_data)
-elif os_name == 'Darwin':
-    lib_data = '{}/darwin64/*.dylib'.format(lib_data)
-else:
-    raise RuntimeError('Could not detect standard (system, architecture).')
+# Readme.
+readme_path = os.path.abspath(os.path.join(os.path.dirname(__file__), 'README.rst'))
+with io.open(readme_path, encoding='utf-8') as f:  # io.open for Python 2 support with encoding
+    README = f.read()
 
 setup(
-    name="buildingspy",
-    version=__version__,
+    name=MAIN_PACKAGE,
+    version=VERSION,
     author="Michael Wetter",
     author_email="mwetter@lbl.gov",
     description=(
         "Package for simulating and testing models from the Modelica Buildings and IBPSA libraries"),
-    long_description=read('README.rst'),
+    long_description=README,
+    long_description_content_type='text/x-rst',
     license="3-clause BSD",
     keywords="modelica dymola openmodelica mat",
     url="http://simulationresearch.lbl.gov/modelica/",
+    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*',
     install_requires=[
-        'future',
-        'gitpython',
-        'jinja2',
-        'matplotlib',
-        'numpy',
-        'pytidylib',
-        'scipy',
-        'simplejson',
-        'six',
+        'future>=0.16',
+        'gitpython>=2.1',
+        'jinja2>=2.10',
+        'matplotlib>=2.2',
+        'numpy>=1.14',
+        'pytidylib>=0.3.2',
+        'scipy>=1.1',
+        'simplejson>=3.14',
+        'six>=1.11',
+        'pyfunnel@git+https://github.com/lbl-srg/funnel.git@master',
```

Comment:
Can these be removed because now there is a requirements.txt file? Also, some versions are different than the ones in requirements.txt.

Reply:
I apply the rule summarized in the §Conclusion [here](https://blog.miguelgrinberg.com/post/the-package-dependency-blues). The idea is to have 

- a loose mechanism to define dependencies with `install_requires` in `setup.py` which is the file used when calling `pip install .`, the minimum version has been chosen to correspond to the release of Ubuntu 18.04,
- an exact mechanism to define dependencies in `requirements.txt` which is the file used for development and testing, and to provide at least a record of a set of specific versions that are known to work with our component. This needs an additional `pip install -r requirements.txt` which I included in `.travis.yml` before `pip install .`. The versions in `requirements.txt` are the ones in my system output typically with `pip freeze > requirements.txt`.

Reply:
I see.